starknet_py/net/full_node_client.py

```diff
@@ -107,20 +111,25 @@ async def get_block(
             method_name="getBlockWithTxs",
             params=block_identifier,
         )
-        return cast(StarknetBlock, StarknetBlockSchema().load(res, unknown=EXCLUDE))
+        if "new_root" in res:
+            return cast(StarknetBlock, StarknetBlockSchema().load(res, unknown=EXCLUDE))
```

Comment:
In theory I think the if condition maybe should be something like 
```
if block_identifier == {"blockNumber": "pending"}:
```
but this also works I guess. wdyt 🤔

Reply:
Your version is more readable, I think